### PR TITLE
LG-3930 Alert on expired cert when walking chain

### PR DESCRIPTION
**Pull request by stevegsa**

### CODE REVIEW

File: app/services/certificate_store.rb [hunk 1]
```diff
@@ -144,4 +145,16 @@ def extract_certs(raw)
...
+  def alert_on_expired_cert(cert)
+    now = Time.zone.now
+    return if cert.not_after >= now
+
+    NewRelic::Agent.notice_error("\
+Certificate Expired. \
+Expiration: #{cert.not_after}, \
+Subject: #{cert.subject}, \
+Issuer: #{cert.issuer}, \
+Key ID: #{cert.key_id}")
```
Comment (zachmargolis) [inline app/services/certificate_store.rb:158]: The trailing slashes are hard to read, what if we use ActiveSupport's `String#squish`?

```suggestion
    NewRelic::Agent.notice_error(<<-STR.squish)
      Certificate Expired.
      Expiration: #{cert.not_after},
      Subject: #{cert.subject},
      Issuer: #{cert.issuer},
      Key ID: #{cert.key_id}
    STR
```

**Review by zachmargolis — APPROVED** [review]
LGTM, some style suggestions

File: app/services/certificate_store.rb (hunk 1 above)
Comment (stevegsa) [reply app/services/certificate_store.rb:158]: nice!

File: app/services/certificate_store.rb [hunk 2]
```diff
@@ -144,4 +145,16 @@ def extract_certs(raw)
       Certificate.new(OpenSSL::X509::Certificate.new(pem + END_CERTIFICATE)) if pem.strip.present?
     end.compact.select(&:ca_capable?)
   end
+
+  def alert_on_expired_cert(cert)
+    now = Time.zone.now
+    return if cert.not_after >= now
```
[the remark this answers is not in the document]
Comment (stevegsa) [reply app/services/certificate_store.rb:151]: there is no expired? method on OpenSSL::X509::Certificate.  I could make a class method on our certificate and use it.  your thoughts?

[the remark this answers is not in the document]
Comment (zachmargolis) [reply app/services/certificate_store.rb:151]: We can reuse the instance method then, `Certificate.new(cert).expired?`

**Pull request merged**